python/lsst/pipe/tasks/multiBand.py

```diff
@@ -818,14 +818,271 @@ def getSkySourceFootprints(self, mergedList, skyInfo, seed):
+            List of data references for each filter
+        """
+        if self.config.simultaneous:
+            # Use SCARLET to simultaneously deblend across filters
+            msg = "Simultaneous deblending will be implemented once DM-15106 is completed"
+            raise NotImplementedError(msg)
+
+            # implement deblender here
+            # ...
+
+            # this is the proper code to write
+            fluxCatalogs, templateCatalogs = self.multibandBeblend.run(exposures, sources)
+            for n in len(patchRefList):
+                self.write(patchRefList[n], fluxCatalogs[n], templateCatalogs[n])
+        else:
+            # Use the singeband deblender to deblend each band separately
+            for patchRef in patchRefList:
+                exposure = patchRef.get(self.config.coaddName + "Coadd_calexp", immediate=True)
+                exposure.getPsf().setCacheCapacity(psfCache)
+                sources = self.readSources(patchRef)
+                self.singleBandDeblend.run(exposure, sources)
+                self.write(patchRef, sources)
+
+    def readSources(self, dataRef):
+        """Read merged catalog
+
+        Read the catalog of merged detections and create a catalog
+        in a single band.
+
+        Parameters
+        ----------
+        dataRef: data reference
+            Data reference for catalog of merged detections
+
+        Returns
+        -------
+        sources: `SourceCatalog`
+            List of sources in merged catalog
+
+        We also need to add columns to hold the measurements we're about to make
+        so we can measure in-place.
+        """
+        merged = dataRef.get(self.config.coaddName + "Coadd_mergeDet", immediate=True)
+        self.log.info("Read %d detections: %s" % (len(merged), dataRef.dataId))
+        idFactory = self.makeIdFactory(dataRef)
+        for s in merged:
+            idFactory.notify(s.getId())
+        table = afwTable.SourceTable.make(self.schema, idFactory)
+        sources = afwTable.SourceCatalog(table)
+        sources.extend(merged, self.schemaMapper)
+        # Fix the testing line below after code review but before merging
+        # (the reviewer is welcome to remind me to do this)
+        return sources[:10] 
+
+    def write(self, dataRef, flux_sources, template_sources=None):
+        """Write the source catalog(s)
+
+        Parameters
+        ----------
+        dataRef: Data Reference
+            Reference to the output catalog.
+        flux_sources: `SourceCatalog`
+            Flux conserved sources to write to file.
+            If using the single band deblender, this is the catalog
+            generated.
+        template_sources: `SourceCatalog`
+            Source catalog using the multiband template models
+            as footprints.
+        """
+        # The multiband deblender does not have to conserve flux,
+        # so only write the flux conserved catalog if it exists
+        if flux_sources is not None:
```

Comment:
True, but I don't think there is any harm in explicitly checking for `None`, since `flux_sources` should either be a `SourceCatalog` or `None` and this is slightly more robust.